### ISSUE TITLE
some questions about the param num_update in optimizer.py

### ISSUE THREAD

**Issue by 501177639** [apache/incubator-mxnet#5059]
Hi,
      I find there are some problems in updating the param num_update in optimizer.py.
For example,
      In the first iter, the first grad gets the lr when the num_update is 0 https://github.com/dmlc/mxnet/blob/master/python/mxnet/optimizer.py#L264 , and then it updates the num_update to 1, so the next grad and others will get the lr  when the num_update is 1,.
     In the second iter, the num_update of the first grad is 1, and others will be 2.
     ........ 
     This may lead to different grads get different lr. It doesn't matter if we use MultiFactorScheduler to get lr https://github.com/dmlc/mxnet/blob/master/python/mxnet/lr_scheduler.py#L85 . However, It does matter if we use other schedulers such as poly scheduler https://github.com/BVLC/caffe/blob/master/src/caffe/solvers/sgd_solver.cpp#L20 .  

    I think the function _update_count should be changed to       https://github.com/dmlc/mxnet/blob/master/python/mxnet/optimizer.py#L136:
         if index not in self._index_update_count:
            self._index_update_count[index] = self.begin_num_update
        self.num_update = self._index_update_count[index]
        self._index_update_count[index] += 1

and there should be changed to  https://github.com/dmlc/mxnet/blob/master/python/mxnet/optimizer.py#L264
        self._update_count(index)
         lr = self._get_lr(index)
        wd = self._get_wd(index)

**Comment by piiswrong** [comment]
Would you like to submit a pull request?

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
